status/status.go

```diff
@@ -0,0 +1,91 @@
+package status
+
+import "sync"
+
+// Status represents a status update from an action within Empire.
+type Status interface {
```

Comment:
yea, i initially had this but was looking at `events` and they implemented an interface. go makes it easy to do things before you need to :)